onnx/common/path.h

```diff
@@ -8,17 +8,67 @@
 #pragma once
 
 #include <string>
+#ifdef _WIN32
+#include <windows.h>
+#endif
 
 namespace ONNX_NAMESPACE {
 
 #ifdef _WIN32
 const std::string k_preferred_path_separator = "\\";
+const std::wstring w_k_preferred_path_separator = L"\\";
 #else // POSIX
 const std::string k_preferred_path_separator = "/";
 #endif
 
-std::string path_join(const std::string& origin, const std::string& append);
-void normalize_separator(std::string& path);
-std::string clean_relative_path(const std::string& path);
+template <typename STRING>
+STRING path_join(const STRING& origin, const STRING& append);
+
+template <typename CHAR>
+bool is_path_separator(CHAR c) {
+  // Windows accept / as path separator.
+  if (k_preferred_path_separator == "\\") {
+    return c == '\\' || c == '/';
+  }
+
+  return c == k_preferred_path_separator[0];
+}
+
+// To always use backward slash as path separator on Windows
+template <typename STRING>
+void normalize_separator(STRING& path) {
```

Comment:
On Windows, please use 
```
path canonical(const path& pval, const path& base = current_path());
path canonical(const path& pval, error_code& ec);
path canonical(const path& pval, const path& base, error_code& ec);
```
from std::filesystem

Reply:
To confirm, are you suggesting that for Windows we should directly use `canonical` instead of using `clean_relative_path` to clean relative path on other platforms?

Reply:
I think existing lexically_normal+make_prefered from filesystem should cover `clean_relative_path` and `normalize_separator` for Windows. I will update this PR soon (remove self implementation and utilize existing functions whenever possible) and let you know. Thanks for the pointer!

Reply:
> To confirm, are you suggesting that for Windows we should directly use `canonical` instead of using `clean_relative_path` to clean relative path on other platforms?

Yes.